### DISCUSSION TITLE
Conditionally render different instances of same action in table method on RelationManager

### DISCUSSION THREAD

**Discussion by joshembling** [filamentphp/filament#5892] [General]
In my table method I need to conditionally render two different cases of `EditActions` - with a different label, model heading, icon, form fields etc. (basically everything).

I thought one way would be to chain `->visible(...condition)` onto `Tables\Actions\EditAction::make()` and then have another `Tables\Actions\EditAction::make()` underneath with different visibility conditions, but this only renders the latest one in the array.

Example here: 
```
public static function table(Table $table): Table
{
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('id'),
                Tables\Columns\TextColumn::make('name'),
            ])
                ->actions([
                Tables\Actions\EditAction::make()->label()->icon()->form()->visible(first condition if they haven't done ABC, render ABC data),
                Tables\Actions\EditAction::make()->label()->icon()->form()->visible(second condition if they've done ABC, render XYZ data), // only recognises this one
}
```

I wouldn't mind rendering two separate return statements from the $table instance (albeit a lot of code), but I don't have access to the Model.

Does anyone have any solutions to what is very simple to do outside of Filament? There will be a lot of different fields in both.

**Comment by craigkuhns** [comment]
I believe each action has to have a unique name assigned to it for them to render correctly. By default the EditAction will have a name of 'edit'. Something like this should work:

```
public static function table(Table $table): Table
{
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('id'),
                Tables\Columns\TextColumn::make('name'),
            ])
                ->actions([
                Tables\Actions\EditAction::make()->label()->icon()->form()->visible(first condition if they haven't done ABC, render ABC data),
                Tables\Actions\EditAction::make('customEditName')->label()->icon()->form()->visible(second condition if they've done ABC, render XYZ data), // only recognises this one
}
```

**Reply by joshembling** [reply]
Amazing Craig, that works like a charm. Don't know how I missed that in the docs but it's here if anyone is curious [https://filamentphp.com/docs/2.x/tables/actions#single-actions](https://filamentphp.com/docs/2.x/tables/actions#single-actions)
